src/Symfony/Bridge/Doctrine/Middleware/DoctrineConnectionMiddleware.php

```diff
@@ -0,0 +1,32 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bridge\Doctrine\Middleware;
+
+use Doctrine\DBAL\Driver;
+use Doctrine\DBAL\Driver\Middleware;
+use Doctrine\Persistence\ManagerRegistry;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Based on https://github.com/Baldinof/roadrunner-bundle/blob/3.x/src/Integration/Doctrine/DoctrineORMMiddleware.php.
+ */
+class DoctrineConnectionMiddleware implements Middleware
+{
+    public function __construct(private ManagerRegistry $managerRegistry, private ContainerInterface $container)
+    {
+    }
+
+    public function wrap(Driver $driver): Driver
+    {
+        return new DoctrineConnectionDriver($driver, $this->managerRegistry, $this->container);
```

Comment:
Let's embed driver here as anonymous class, saves us a headache naming it and this is how traditional dbal middlewares are done (see `\Doctrine\DBAL\Driver\AbstractSQLiteDriver\Middleware\EnableForeignKeys`, `\Doctrine\DBAL\Driver\OCI8\Middleware\InitializeSession`...)